### PR TITLE
Issue #328: fix for validationType for non base token properties

### PR DESCRIPTION
**Pull request by gaurabdg**
Issue #328 
### DescendantTokenCheck
Screenshot 2020-09-05 at 9 29 24 PM
execution:
Screenshot 2020-09-12 at 4 09 09 PM

### MultipleStringLiteralsCheck
Screenshot 2020-09-05 at 9 32 02 PM
execution:
Screenshot 2020-09-06 at 4 21 20 AM

### MagicNumberCheck
Screenshot 2020-09-05 at 10 03 16 PM
execution:
Screenshot 2020-09-06 at 4 48 53 AM

### IllegalTypeCheck
Screenshot 2020-09-05 at 10 09 22 PM
execution:
Screenshot 2020-09-06 at 5 37 06 PM

### AtclauseOrderCheck
Screenshot 2020-09-05 at 3 53 41 AM
execution:
Screenshot 2020-09-06 at 6 44 30 PM

### CODE REVIEW

**Comment by romani** [comment]
https://checkstyle.org/config_misc.html#DescendantToken

`tokens` is not referenced in javadoc and xdoc.
We usually have such model ONLY for cases when `tokens` is not configurable (getRequredToekns == getAcceptableTokens).
but this is different case, we just need all tokens. We might did not wanted to list all tokens, as it is too much we were too lazy.
Lets do some HACK to unblock us:
please introduce `tokens` to javadoc and make new type `all tokens TokenTypes`
and please make all required hacks in UTs to make them appear in metafiles as list of all tokens or just as type `allTokenTypes` to let plugins treat them in special way.

**Review by muhlba91 — APPROVED** [review]

**Comment by muhlba91** [comment]
please check build failures :)

File: pom.xml [hunk 1]
```diff
@@ -93,7 +93,7 @@
   </ciManagement>
 
   <properties>
-    <checkstyle.version>8.36</checkstyle.version>
+    <checkstyle.version>8.37-SNAPSHOT</checkstyle.version>
```
Comment (romani) [inline pom.xml:96]: please use 8.36.1, it is released

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: pom.xml (hunk 1 above)
Comment (gaurabdg) [reply pom.xml]: Done

**Comment by romani** [comment]
Wercker is failed in sonar validation
`Cloning into 'blog-sonar-build-breaker'...
Timeout of  + 36 +  seconds exceeded`

The same problem is in checkstyle repo, I am not sure why ...

**Pull request merged**